### DISCUSSION TITLE
Updating from an old version help

### DISCUSSION THREAD

**Discussion by multiaki** [vmware-clarity/ng-clarity#331] [Q&A]
Hello

I am trying to upgrade an angular project from v0.13. I am seeing new class being added body p:not([cds-text]) that overwrites my css styles. Is there a place I can follow to properly upgrade the Clarity design. 

Thanks

**Comment by kevinbuhmann** [comment]
I think the best we can offer are the old changelogs you can find them here: https://clarity.design/news.

v0.13 is VERY old version, so you will likely encounter many breaking changes on the way to v13.x.

**Comment by multiaki** [comment]
@kevinbuhmann thanks for the response. Do I need to do anything special to get rid of "body p:not([cds-text])" I think this is added by default.

**Reply by kevinbuhmann** [reply]
There isn't a way to remove parts of the Clarity styles, but you can probably adjust your application styles to fix whatever the issue is.

I would make sure your `p` styles have higher CSS specificity than the Clarity styles. Make sure your styles are _after_ the Clarity styles, increase the selector specificity if needed, and add rules to undo whatever Clarity is doing that you don't want.

However, the styles you mention are part of Clarity's global typography rules, and I'm not sure how much customization should be done if you want to stay aligned with the design system. You can do whatever you need in your application styles as outlined above, though. This is just something to keep in mind.

**Comment by multiaki** [comment]
I looked at the issue. Here is the issue.
I used to have a project where material UI css was coming first. Now after adding "h2:not([cds-text])", the css went on top of the material UI theme. Is there a way to fix it?

**Reply by kevinbuhmann** [reply]
I'm not sure what you're asking.

**Reply by ashleyryan** [reply]
You'll have to tweak your material theme or find ways to customize it to increase specificity. You're right that the clr/ui styles were updated with `not:([cds-*])` to avoid collisions with Clarity Core, which increased the specificity of the selectors and apparently made them beat out the material CSS, but unfortunately specificity issues are something you'll run into when using two globally-scoped component libraries together
